src/components/Login/Login.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import Input from '../../common/Input/Input';
+import Button from '../../common/Button/Button';
+import './Login.css';
+
+const Login = ({ setUserName }) => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const navigate = useNavigate();
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const response = await fetch('http://localhost:4000/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, password }),
+      });
+
+      if (response.ok) {
```

Comment:
To eliminate a lot if `if-else` branching, we could re-write this vice-versa and make a bit more streamlined:

```
if (!response.ok) {
  console.error(...);
  return;
}

if (!result.successful || !result.result) {
  console.error(...);
  return;
}

// ... do all the token related functionality ...
```

Reply:
Instead of:

```
if 
  if 
    if
    else
  else
else
```

We have this:
```
if NOT
  return
  
if NOT
  return
  
else
```